api/api.go

```diff
@@ -47,6 +47,20 @@ type API interface {
 	GetMetricsForTag(tagKey, tagValue string) ([]MetricKey, error)
 }
 
+type APIGraphiteStore interface {
+	// Embed the API interface
+	API
+
+	// Add a Graphite metric to the complete list (as needed)
+	AddGraphiteMetric(metric GraphiteMetric) error
+
+	// Get all the Graphite metrics
+	GetAllGraphiteMetrics() ([]GraphiteMetric, error)
+
+	// Checks whether (at runtime) the object actually is capable of performing these queries
+	SupportsGraphiteStore() bool
```

Comment:
i think this feels a bit weird, since individual methods are capable of returning 'not supported error'.